glean-core/src/storage/mod.rs

```diff
@@ -42,6 +42,33 @@ fn snapshot_labeled_metrics(
     obj.insert(label.into(), metric.as_json());
 }
 
+/// Labeled metrics are stored as `<metric id>/<label>`.
+/// They need to go into a nested object in the final snapshot.
+///
+/// We therefore extract the metric id and the label from the key and construct the new object or
+/// add to it.
+fn snapshot_dual_labeled_metrics(
+    snapshot: &mut HashMap<String, HashMap<String, JsonValue>>,
+    metric_id: &str,
+    metric: &Metric,
+) {
+    let ping_section = format!("dual_labeled_{}", metric.ping_section());
+    let map = snapshot.entry(ping_section).or_default();
+
+    // Safe unwrap, the function is only called when the id does contain a '/'
+    let parts = metric_id.split('\x1e').collect::<Vec<&str>>();
+
+    let obj = map
+        .entry(parts[0].into())
+        .or_insert_with(|| json!({}))
+        .as_object_mut()
+        .unwrap();
+    // safe unwrap, we constructed the object above
+    let key_obj = obj.entry(parts[1].to_string()).or_insert_with(|| json!({}));
+    let key_obj = key_obj.as_object_mut().unwrap();
+    key_obj.insert(parts[2].into(), metric.as_json());
```

Comment:
We should have _some_ check to ensure we actually have 2 separators and thus 2 parts in the id.
The function below only checks that it contains at least one.